Zend/zend_exceptions.c

```diff
@@ -800,6 +801,10 @@ void zend_register_default_exception(void) /* {{{ */
 	zend_ce_division_by_zero_error->create_object = zend_default_exception_new;
 
 	INIT_CLASS_ENTRY(zend_ce_unwind_exit, "UnwindExit", NULL);
+
+	INIT_CLASS_ENTRY(ce, "UnhandledMatchError", NULL);
+	zend_ce_unhandled_match_error = zend_register_internal_class_ex(&ce, zend_ce_error);
```

Comment:
Extend `ValueError`?

Reply:
No preference. @kocsismate ?

Reply:
No hard preference for me as well, but probably I'd leave it as-is. If I had to give an explanation for this, I'd say that `ValueError`s are used when the value of an argument is not right, but in this case we can't be sure that the value itself is problematic or there's a missing arm.

Reply:
Ok, we'll leave it as is then. 🙂